### PR TITLE
Merge the Spring Cloud Azure 4.1.0 release changes back to main

### PR DESCRIPTION
**Pull request by yiliuTo**
# Description
This pr is to merge the spring cloud azure 4.1.0 release branch to main branch.

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [x] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).**

### CODE REVIEW

File: sdk/spring-experimental/spring-cloud-azure-native-configuration/pom.xml [hunk 1]
```diff
@@ -144,6 +144,25 @@
           </rules>
         </configuration>
       </plugin>
+      <!-- We must generate an empty javadoc JAR file to publish to Maven Central because this package does not contain any public sources -->
```
Comment (chenrujun) [inline sdk/spring-experimental/spring-cloud-azure-native-configuration/pom.xml:147]: Is this required by maven central or release pipeline?

**Review by chenrujun — APPROVED** [review]
LGTM.

File: sdk/spring-experimental/spring-cloud-azure-native-configuration/pom.xml (hunk 1 above)
Comment (yiliuTo) [reply sdk/spring-experimental/spring-cloud-azure-native-configuration/pom.xml:147]: Yeah, we added this to pass the Maven central rules of release packages.

**Pull request merged**